python/lsst/ci/hsc/validate.py

```diff
@@ -4,15 +4,51 @@
 
 import os
 import numpy
+import argparse
+from lsst.base import setNumThreads
 from lsst.pex.logging import getDefaultLog
 from lsst.daf.persistence import Butler
 from lsst.meas.astrom import LoadAstrometryNetObjectsTask
 
-_butler = {}
-def getButler(root):
-    if not root in _butler:
-        _butler[root] = Butler(root)
-    return butler[root]
+
+class IdValueAction(argparse.Action):
+    """argparse action callback to process a data ID
+
+    We don't support as full a range of operators as does the pipe_base ArgumentParser.
```

Comment:
Can you expand on what this is actually supposed to do?

From the help message (`--id [KEY=VALUE1[^VALUE2[^VALUE3...`), it looks as if it supports the usual data ID syntax for specifying multiple values. In practice, it doesn't: trying to specify multiple CCDs will bomb out with something like `ValueError: invalid literal for int() with base 10: '1^2^3'`. 

Is that what the "don't support as full a range of operators" comment is referring to? If so, please make it more explicit.